### ISSUE TITLE
Add a diff functionality for Tab and JSON files

### ISSUE THREAD

**Issue by agbeltran** [ISA-tools/isa-api#139]
I think it would be useful to have a diff-like functionality for ISA tab and JSON files, so that we can compare different versions of the same experiment description. This functionality will highlight the differences in the experiment description element by element.

**Comment by djcomlab** [comment]
I already started on this is already present for JSON, see https://github.com/ISA-tools/isa-api/blob/enhance/tests/tests/utils.py

If `assert_json_equal()` returns False it also uses `DeepDiff` to report the diff.

if `assert_tab_content_equal()` returns False, `pandas.util.testing.assert_frame_equal()` report a diff.

Take a look in https://github.com/ISA-tools/isa-api/blob/enhance/tests/tests/test_isatab2json.py and https://github.com/ISA-tools/isa-api/blob/enhance/tests/tests/test_json2isatab.py for usages.

It's not too easy to diff ISA content as most diff utilities will report a diff if for example column order is different (where actually in some cases column order doesn't matter in ISA tab), and in some JSON and XML lists order is not automatically guaranteed. I made a function in `tests.utils`, `immutablesort(json)` that seems to work well enough on sorting JSON consistently and need to work on one for XML content for testing SRA export.
